src/main/kotlin/uk/gov/dwp/dataworks/services/TaskDeploymentService.kt

```diff
@@ -73,11 +87,19 @@ class TaskDeploymentService {
                     .build()
             awsCommunicator.createAlbRoutingRule(correlationId, userName, listener.listenerArn(), targetGroup.targetGroupArn())
 
-            // IAM permissions
-            parsePolicyDocuments(additionalPermissions)
-            val iamPolicy = awsCommunicator.createIamPolicy(correlationId, userName, taskRolePolicyString)
-            val iamRole = awsCommunicator.createIamRole(correlationId, userName, taskAssumeRoleString)
+            //IAM Permissions
+            val sharedFolder = awsParsing.createArnStringsList(cognitoGroups, "shared", "${configurationResolver.getStringConfig(ConfigKey.JUPYTER_S3_ARN)}/*")
+            val userFolder = awsParsing.createArnStringsList(listOf(userName), "home", configurationResolver.getStringConfig(ConfigKey.JUPYTER_S3_ARN))
+            val folderAccess = sharedFolder.plus(userFolder)
+            val mapForParsing = mapOf(Pair("jupyter-s3-access-document", folderAccess), Pair("jupyter-s3-list", listOf(configurationResolver.getStringConfig(ConfigKey.JUPYTER_S3_ARN))))
+            jupyterBucketAccessRolePolicyString = awsParsing.parsePolicyDocument(jupyterBucketAccessDocument, mapForParsing, "Resources")
+            taskRolePolicyString = awsParsing.parsePolicyDocument(taskRolePolicyDocument, mapOf("ecs-task-role-policy" to additionalPermissions), "Actions")
+            taskAssumeRoleString = taskAssumeRoleDocument.inputStream.bufferedReader().use { it.readText() }
+            val jupyterIamPolicy = awsCommunicator.createIamPolicy(correlationId, "$userName-jupyter-s3-document", jupyterBucketAccessRolePolicyString)
+            val iamPolicy = awsCommunicator.createIamPolicy(correlationId, "$userName-task-role-document", taskRolePolicyString)
+            val iamRole = awsCommunicator.createIamRole(correlationId, "$userName-iam-role", taskAssumeRoleString)
```

Comment:
I feel like everything `jupyter` related here bulks out this method to an unreadable level and things like `sharedFolder` could do with some context that it is relating to jupyter. Can we break out all that to separate method please?

Reply:
I see that you resolved my comment but I don't see a change - was this lost in the mists of git rebasing?

Reply:
This is what the parseMap function is - it has reduced it to the policy strings, policies, role and attaching policies to roles - all of which are directly relevant to the IAM Permissions subheading. I think that this reads a lot better now and contains the relevant steps to create IAM policies, so that this can all be changed in one place, if needed.